controller/eks-cluster-config-handler.go

```diff
@@ -578,6 +555,27 @@ func (h *Handler) validateCreate(config *eksv1.EKSClusterConfig, eksService *eks
 	for _, ng := range config.Spec.NodeGroups {
 		cannotBeNilError := "field [%s] cannot be nil for nodegroup [%s] in non-nil cluster [%s]"
 		if !config.Spec.Imported {
+			if ng.LaunchTemplate != nil {
+				if ng.LaunchTemplate.ID == nil {
+					return fmt.Errorf(cannotBeNilError, "launchTemplate.ID", *ng.NodegroupName, config.Name)
+				}
+				if ng.LaunchTemplate.Version == nil {
+					return fmt.Errorf(cannotBeNilError, "launchTemplate.Version", *ng.NodegroupName, config.Name)
+				}
+			} else {
```

Comment:
Is this else block needed? Do you want to check if the fields are not nil regardless of the conditional?

Reply:
If the node group has a launch template, then the values in the `else` block are taken care of in the launch template. If the node group doesn't have a launch template, then we need to check these values.